llvm/lib/Target/RISCV/MCTargetDesc/RISCVTargetStreamer.cpp

```diff
@@ -75,6 +75,13 @@ void RISCVTargetStreamer::emitTargetAttributes(const MCSubtargetInfo &STI,
     auto &ISAInfo = *ParseResult;
     emitTextAttribute(RISCVAttrs::ARCH, ISAInfo->toString());
   }
+
+  if (STI.hasFeature(RISCV::FeatureStdExtA)) {
+    unsigned AtomicABITag = STI.hasFeature(RISCV::FeatureTrailingSeqCstFence)
+                                ? RISCVAttrs::RISCVAtomicAbiTag::AtomicABI::A6S
```

Comment:
Is this the right way round? Isn't `STI.hasFeature(RISCV::FeatureTrailingSeqCstFence)` true if `+no-seq-cst-trailing-fence` is present? And if there's no trailing fence, then we have A6C? Or am I misreading?

I think it would be better to name the feature `FeatureNoTrailingSeqCstFence` as we do for `FeatureNoRVCHints`. Though I think this patch would be best if it aimed to change no defaults at all, and left that as a follow-on patch (which would hopefully be a quick LGTM as I think there's no objection to that, going by this review thread).

Reply:
> I think this should be correct. The default is now "enabled", so adding `+no-seq-cst-trailing-fence` should make `STI.hasFeature(RISCV::FeatureTrailingSeqCstFence)` return `false`. I think the tests indicate that's how its working.

So in attributes.ll, the A6S output has the RUN line `; RUN: llc -mtriple=riscv64 -mattr=+a,+no-seq-cst-trailing-fence %s -o - | FileCheck --check-prefixes=CHECK,RV64A,A6S %s`. But for A6S, don't we require that the trailing fence is emitted?

Reply:
huh, you're right. Let me double check that now that the default is in it's own change.